airsonic-main/src/main/java/org/airsonic/player/service/MediaFileService.java

```diff
@@ -196,21 +196,19 @@ private MediaFile checkLastModified(MediaFile mediaFile, boolean useFastCache) {
         List<MediaFile> result = new ArrayList<MediaFile>();
         for (MediaFile child : mediaFileDao.getChildrenOf(parent.getPath())) {
             child = checkLastModified(child, useFastCache);
-            if (child.isDirectory() && includeDirectories && includeMediaFile(child)) {
-                result.add(child);
-            }
-            if (child.isFile() && includeFiles && includeMediaFile(child)) {
-                result.add(child);
+            if (includeMediaFile(child)) {
+                if (child.isDirectory() && includeDirectories) {
```

Comment:
I think the reason I didn't do this initially was that for efficiency the lighter weight boolean check should be before the much heavier regex and string manipulations, since I think this is more often called with either files or dirs enabled but not both.

Reply:
After this commit, there is only one callsite, with files and dirs set to true.

Reply:
If #1322 is merged, I think that there won't be any performance-critical callsite not using both files and dirs set to true.